packages/node_modules/@webex/internal-plugin-device/src/features/feature-model.js

```diff
@@ -79,6 +79,57 @@ const FeatureModel = AmpState.extend({
 
   // Ampsersand method members.
 
+  /**
+   * Parse all attribute values and map them to their appropriate members.
+   *
+   * @param {FeatureModel} pto - Attributes to be parsed and mapped.
+   * @returns {FeatureModel} - Parsed and mapped attributes.
+   */
+  parse(pto) {
+    // Generate a return transfer object.
+    let rto = {};
+
+    // Validate that a parameter transfer object was provided.
+    if (!pto) {
+      return rto;
+    }
+
+    // Clone the parameter transfer object into the return transfer object.
+    rto = {...pto};
+
+    // Validate that a value was provided.
+    if (rto.val) {
```

Comment:
I'm a little confused by this line. If the expected parameter type is `FeatureModel` and the return type is _also_ a `FeatureModel`, why are the attributes different?

Reply:
This method normalizes the feature model DTO. This is also a Ampersand method override that passes itself into the method to normalize data types. I would personally change this if ampersand wasn't required to maintain functionality across the JS SDK. see [this documentation entry](https://ampersandjs.com/docs/#ampersand-state-parse) for more details.

Reply:
So I guess my question is (for context), why do we have to do this normalization in the first place?

Reply:
The data transfer object is received from the `wdm` service as a `Record<string, string>` for features. This makes things appear as so:
```js
{
  // ...
  type: 'boolean'
  val: 'true'
  // ...
}
```
This makes comparison operations like so:
```js
if (feature) { /* ... */ }
```
always true. These features must be re-casted as their proper types.